onnxruntime/core/providers/shared_library/provider_host_api.h

```diff
@@ -11,6 +13,12 @@ struct Provider {
   virtual std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory(int /*device_id*/) { return nullptr; }
 
   virtual void* GetInfo() { return nullptr; }  // Returns a provider specific information interface if it exists
+
+  virtual ProviderOptions GetProviderOptions() { return {}; }  // Returns a provider options interface if it exists
```

Comment:
Should we be returning this by value?

Reply:
https://github.com/microsoft/onnxruntime/blob/master/include/onnxruntime/core/framework/execution_provider.h#L121
The GetProviderOptions() in IExecutionProvider is also return ProviderOptions.